extensions/script-libraries/drawing/drawing.livecodescript

```diff
@@ -0,0 +1,2960 @@
+				put -tVpHeight * tHeight into tVpHeight
+			end if
+
+			/* If the new vp sizes are not percentages then we can build a matrix
+			 * else we must emit a separate transform */
+			if tVpWidth >= 0 and tVpHeight >= 0 then
+				local tVScaleX, tVScaleY, tVTranslateX, tVTranslateY
+				put tVpWidth / tVbWidth into tVScaleX
+				put tVpHeight / tVbHeight into tVScaleY
+				if word 2 of tPar is "meet" then
+					put min(tVScaleX, tVScaleY) into tVScaleX
+					put tVScaleX into tVScaleY
+				else if word 2 of tPar is "slice" then
+					put max(tVScaleX, tVScaleY) into tVScaleX
+					put tVScaleX into tVScaleY
+				end if
+				put tVpX - (tVbX * tVScaleX) into tVTranslateX
+				put tVpY - (tVbY * tVScaleY) into tVTranslateY
+				if tPar contains "xMid" then
+					add (tVpWidth - tVbWidth * tVScaleX) / 2 to tVTranslateX
+				end if
+				if tPar contains "xMax" then
+					add (tVpWidth - tVbWidth * tVScaleX) to tVTranslateX
+				end if
+				if tPar contains "yMid" then
+					add (tVpHeight - tVbHeight * tVScaleY) / 2 to tVTranslateY
+				end if
+				if tPar contains "yMax" then
+					add (tVpHeight - tVbHeight * tVScaleY) to tVTranslateY
+				end if
+				
+				put _svgTransformTranslate(tVTranslateX, tVTranslateY) into tNextMatrix
+				put _svgTransformConcat(tNextMatrix, _svgTransformScale(tVScaleX, tVScaleY)) into tNextMatrix
+			else
+				if not _svgTransformIsIdentity(tMatrix) then
+					seqPushOntoBack rCompiledTransform, _svgTransformFlatten(tMatrix)
+				end if
+				put tVpWidth into tTransform[4]
+				put tVpHeight into tTransform[5]
+				seqPushOntoBack rCompiledTransform, tTransform
+				put _svgTransformIdentity() into tMatrix
+				put _svgTransformIdentity() into tNextMatrix
+			end if
+
+			put tVpWidth into tWidth
+			put tVpHeight into tHeight
+			break
+		case "matrix"
+			put _svgTransformMatrix(tTransform[2], tTransform[3], tTransform[4], tTransform[5], tTransform[6], tTransform[7]) into tNextMatrix
+			break
+		case "translate"
+			put _svgTransformTranslate(tTransform[2], tTransform[3]) into tNextMatrix
+			break
+		case "scale"
+			put _svgTransformScale(tTransform[2], tTransform[3]) into tNextMatrix
+			break
+		case "rotate"
+			put _svgTransformTranslate(-tTransform[3], -tTransform[4]) into tNextMatrix
+			put _svgTransformConcat(tNextMatrix, _svgTransformRotate(tTransform[2])) into tNextMatrix
+			put _svgTransformConcat(tNextMatrix, _svgTransformTranslate(tTransform[3], tTransform[4])) into tNextMatrix
+			break
+		case "skewX"
+			put _svgTransformSkew(tTransform[2], 0) into tNextMatrix
+			break
+		case "skewY"
+			put _svgTransformSkew(0, tTransform[2]) into tNextMatrix
+			break
+		default
+			_InternalError format("unknown transform '%s'", pTransform[1])
+			break
+		end switch
+		put _svgTransformConcat(tMatrix, tNextMatrix) into tMatrix 
+	end repeat
+	if not _svgTransformIsIdentity(tMatrix) then
+		seqPushOntoBack rCompiledTransform, _svgTransformFlatten(tMatrix)
+	end if
+end _svgCompileTransform
```

Comment:
This is maybe too time consuming to do right now, but it would be nice if all the transform array indices were constants with appropriate names

Reply:
Yes - I'll think on that - I think I'd prefer to make the operation arrays dictionaries rather than sequences (so the keys are named), and then map those to the EncodeOp. Perhaps something for a subsequent iteration :)